src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3758,6 +3758,11 @@ protected override void VisitStatement(BoundStatement statement)
             {
                 switch (element)
                 {
+                    case BoundUnconvertedCollectionArguments collectionArguments:
+                        // We only get here if the conversion of the collection expression to the
+                        // target type failed. In this case, simply visit each argument.
+                        VisitArgumentsEvaluate(collectionArguments.Arguments, collectionArguments.ArgumentRefKindsOpt, parameterAnnotationsOpt: default, defaultArguments: default);
+                        break;
```

Comment:
Consider leaving a PROTOTYPE note or tracking a follow-up elsewhere to implement nullability analysis on happy cases (where the conversion succeeds)

Reply:
Added an item to the test plan: [https://github.com/dotnet/roslyn/issues/76310](https://github.com/dotnet/roslyn/issues/76310)